### DISCUSSION TITLE
Allow optionally marking a custom event as non-interactive

### DISCUSSION THREAD

**Discussion by rodrigotassinari** [plausible/analytics#3105] [Anything Else]
We're experimenting with Plausible at my company (an ecommerce website), and starting to use the [custom events](https://plausible.io/docs/custom-event-goals) feature (especially useful now with the recent addition of [revenue tracking](https://plausible.io/docs/ecommerce-revenue-tracking)).

At the [metrics definition docs](https://plausible.io/docs/metrics-definitions) there's a remark on the [bounce rate explanation](https://plausible.io/docs/metrics-definitions#bounce-rate) that "_all custom events are considered interactive_". So this means all custom events either extend the visitor's current visit or create a new one.

We would like to track some actions that, even though were initiated by the visitor, occurred some time after the user has left the site. A major example is purchase tracking when the payment notification may arrive hours or even days after the purchase (at this time, 25% of our sales are paid in this asynchronous fashion).

So if we send this purchase completed event to Plausible we will have an artificially higher number of visits and/or visit duration. If we don't send this event, we won't have precise revenue amount tracking on Plausible.

However if we could mark this event as non-interactive somehow, so as to not be included on visit / session calculations, but still be used for conversion tracking, we could send the event without harming the number of visits and it's durations. I believe Google Analytics v3 has [a similar option](https://www.optimizesmart.com/non-interaction-events-in-google-analytics-explained/).

**Comment by metmarkosaric** [comment]
thanks @rodrigotassinari! makes sense. there's a similar thread here https://github.com/plausible/analytics/discussions/1914

**Reply by rodrigotassinari** [reply]
ah, sorry I didn't find the previous similar thread @metmarkosaric please feel free to merge them or close one of them if desired.

i've also opened [another thread](https://github.com/plausible/analytics/discussions/3076) that also would solve our problem but in a different manner, and would allow back-filling of past events via the API. i'm not sure, from Plausible's point of view, which approach would be best, though.

**Comment by metmarkosaric** [comment]
This is now done. When triggering events using Plausible tracker, you can do `plausible('Custom Event', {interactive: false})`. Via the Events API there’s now an optional `interactive` parameter.
